src/main/java/com/puppycrawl/tools/checkstyle/DefaultLogger.java

```diff
@@ -211,4 +214,94 @@ private void closeStreams() {
         }
     }
 
+    /**
+     * Represents a message that can be localised. The translations come from
+     * message.properties files. The underlying implementation uses
+     * java.text.MessageFormat.
+     *
+     * @noinspection SerializableHasSerializationMethods
+     */
+    /* package */ static class LocalizedMessage {
+
+        /**
+         * A cache that maps bundle names to ResourceBundles.
+         * Avoids repetitive calls to ResourceBundle.getBundle().
+         */
+        private static final Map<String, ResourceBundle> BUNDLE_CACHE =
+                Collections.synchronizedMap(new HashMap<>());
+
+        /**
+         * The locale to localise messages to.
+         **/
+        private static Locale locale = Locale.getDefault();
+
+        /**
+         * Key for the message format.
+         **/
+        private final String key;
+
+        /**
+         * Arguments for MessageFormat.
+         *
+         * @noinspection NonSerializableFieldInSerializableClass
+         */
+        private final Object[] args;
+
+        /**
+         * Creates a new {@code LocalizedMessage} instance.
+         *
+         * @param args arguments for the translation.
+         * @param key the key to locate the translation.
+         * @noinspection ConstructorWithTooManyParameters.
+         */
+        // -@cs[ParameterNumber] Class is immutable, we need that amount of arguments.
+        /* package */LocalizedMessage(Object[] args, String key) {
```

Comment:
violation is valid
if guide you to swap parameters.

Reply:
Swapped parameters. Done.